raft-tech/TANF-app
# Fix line ending inconsistencies for Zap Scan HTML Report markdown

### Summary of Changes
Merging this should hopefully fix the constant re-appearance of `docs/Technical-Documentation/Zap-Scan-HTML-Report.md` in git diffs. It appears that this file was committed with Windows line endings:
```
❯ file docs/Technical-Documentation/Zap-Scan-HTML-Report.md                                                                                                                                                     ─╯
docs/Technical-Documentation/Zap-Scan-HTML-Report.md: ASCII text, with CRLF line terminators
```
Most of us use Mac or Linux, and I assume have set `git config --global core.autocrlf input` - however, in our .gitattributes we had that value set to `auto` which overrides whatever the user setting is. [ref](https://docs.github.com/en/get-started/getting-started-with-git/configuring-git-to-handle-line-endings#per-repository-settings) This setting is the Windows friendly default, and lets git handle line endings how it sees fit - but in our case this conflicts with other declarations in .gitattributes that explicitly set files to use `LF` line endings. If we instead use `input` it will be converted to `LF` only on commit and then on checkout the line endings will be processed using the `core.eol` setting which defaults to `native` and will be handled as needed per OS. [Additional ref](https://adaptivepatchwork.com/2012/03/01/mind-the-end-of-your-line/#:~:text=your%20working%20directory.-,core.,file%20to%20the%20object%20database.)

To convert the file, I ran the following locally and then committed the change:
```
❯ dos2unix docs/Technical-Documentation/Zap-Scan-HTML-Report.md                                                                                                                                                 ─╯
dos2unix: converting file docs/Technical-Documentation/Zap-Scan-HTML-Report.md to Unix format...
```

### How to Test
* Check out this branch on your local machine
* Confirm that `docs/Technical-Documentation/Zap-Scan-HTML-Report.md` does not show as a changed file and has the correct line endings:
```
❯ file docs/Technical-Documentation/Zap-Scan-HTML-Report.md                                                                                                                                                     ─╯
docs/Technical-Documentation/Zap-Scan-HTML-Report.md: ASCII text
```

### Deliverable 1: Accepted Features

> *Performance Standard(s): At the beginning of each sprint, the Product Owner and development team will collaborate to define a set of user stories to be completed during the sprint. Acceptance criteria for each story will also be defined. The development team will deliver code and functionality to satisfy these user stories.*

> *Acceptable Quality Level: Delivered code meets the acceptance criteria for each user story. Incomplete stories will be assessed and considered for inclusion in the next sprint.*

  - [x] Zap scan doc does not show as a changed file due to line ending differences on checkout

As facilitator/product manager, @kniz-raft will decide if ACs are met from Raft's perspective.

### Deliverable 2: Tested Code

> *Performance Standard(s): Code delivered under the order must have substantial test code coverage. Version-controlled HHS GitHub repository of code that comprises products that will remain in the government domain.*

> *Acceptable Quality Level: Minimum of 90% test coverage of all code. All areas of code are meaningfully tested.*

N/A repo config change

### Deliverable 3: Properly Styled Code

> *Performance Standard(s): GSA 18F Front- End Guide*

> *Acceptable Quality Level: 0 linting errors and 0 warnings*

N/A repo config change

### Deliverable 4: Accessible

> *Performance Standard(s): Web Content Accessibility Guidelines 2.1 AA standards*

> *Acceptable Quality Level: 0 errors reported using an automated scanner and 0 errors reported in manual testing*

N/A repo config change

### Deliverable 5: Deployed

> *Performance Standard(s): Code must successfully build and deploy into the staging environment.*

> *Acceptable Quality Level: Successful build with a single command*

N/A repo config change

### Deliverable 6: Documented

> *Performance Standard(s): Summary of user stories completed every two weeks. All dependencies are listed and the licenses are documented. Major functionality in the software/source code is documented, including system diagram. Individual methods are documented inline in a format that permits the use of tools such as JSDoc. All non-inherited 800-53 system security controls are documented in the Open Control or OSCAL format and HHS Section 508 Product Assessment Template (PAT) are updated as appropriate.*

> *Acceptable Quality Level: Combination of manual review and automated testing, if available*

N/A repo config change

### Deliverable 7: Secure

> *Performance Standard(s): Open Web Application Security Project (OWASP) Application Security Verification Standard 3.0*

> *Acceptable Quality Level: Code submitted must be free of medium- and high-level static and dynamic security vulnerabilities*

N/A repo config change
